src/java/htsjdk/samtools/reference/IndexedFastaSequenceFile.java

```diff
@@ -202,10 +251,20 @@ public ReferenceSequence getSubsequenceAt( String contig, long start, long stop
             startOffset += Math.max((int)(startOffset%bytesPerLine - basesPerLine + 1),0);
 
             try {
-                 startOffset += channel.read(channelBuffer,indexEntry.getLocation()+startOffset);
+                if (channel instanceof FileChannel) {
+                    startOffset += ((FileChannel) channel).read(channelBuffer,indexEntry.getLocation()+startOffset);
```

Comment:
Agree with @lbergelson - in fact doesn't this change the semantics of the operation depending on whether it's a FileChannel or not?  If it's a FileChannel you end up with the channel pointing at the end of the last read, but if it's any other channel it gets reset back to the prior position.  Seems undesirable to have differing behaviour based on the underlying kind of channel